### PR TITLE
add citydao-one-human-one-vote strategy [citydao-one-human-one-vote]

### PR DESCRIPTION
**Pull request by RosebuDAO**
address must be in proof of humanity registry.
address must hold at least 1 CityDAO Citizen NFT,
granting them a maximum of 1 vote

### CODE REVIEW

**Comment by RosebuDAO** [comment]
haven't tested this strategy at all but seeking feedback. am I even doing it right?

I am trying to find an address that is in PoH registry and owns a Citizen NFT, will force push updating examples.json once I have one

**Review by ChaituVR — CHANGES_REQUESTED** [review]
```
Error: src/strategies/citydao/index.ts(31,27): error TS2551: Property 'foreach' does not exist on type '[string, number][]'. Did you mean 'forEach'?
Error: src/strategies/citydao/index.ts(38,13): error TS2349: This expression is not callable.
```

**Comment by RosebuDAO** [comment]
As for testing, Vitalik Buterin is in the POH registry with this address: 0x1db3439a222c519ab44bb1144fc28167b4fa6ee6 and he owns [several erc1155 NFTs](https://etherscan.io/address/0x1db3439a222c519ab44bb1144fc28167b4fa6ee6#tokentxnsErc1155)
here's one: 0x179335e8b0005210c2fa9a1dfb7f92cf636157ad
tokenId: 140

still trying to find a citydao citizen with proof of humanity though

**Comment by ChaituVR** [comment]
@RosebuDAO 
There are still errors:
```js
src/strategies/citydao/index.ts(38,13): error TS[23](https://github.com/snapshot-labs/snapshot-strategies/runs/6294108347?check_suite_focus=true#step:3:23)49: This expression is not callable.
  Type 'Number' has no call signatures.
error Command failed with exit code 2.
```
Try running this in your local:
```bash
yarn test --strategy=citydao --more=500
```


and for 
> still trying to find a citydao citizen with proof of humanity though

May be try this https://snapshot.org/#/strategy/proof-of-humanity

**Comment by RosebuDAO** [comment]
I got a test env set up and did some research but can't figure out how to solve that bug. Maybe the code is incorrect in the first place? My experience with JS/TS is minimal.

`const result1: Record<string, number> = await multi1.execute();`

how do I convert this to an object that my strategy can return?

File: src/strategies/citydao/index.ts
Comment (RosebuDAO) [inline src/strategies/citydao/index.ts:44]: Output of this console.log:
Map(0) {
      '0xf4d895311987faebd839908a0793461087b229c6': 0,
      '0x1db3439a222c519ab44bb1144fc28167b4fa6ee6': 1
    }


Really puzzled why this strategy isn't working at this point. The test suite fails on "Should return an array of object with addresses". It is seeing an empty object where it shouldn't.

Comment (RosebuDAO) [reply src/strategies/citydao/index.ts:44]: it's because I was using votes[address] = balance; .... instead of votes.set(address, balance);

File: src/strategies/citydao/examples.json
Comment (RosebuDAO) [inline src/strategies/citydao/examples.json:8]: This isn't the CityDAO Citizen NFT it's just some random erc1155 that Vitalik Buterin happens to hold

**Review by RosebuDAO — COMMENTED** [review]
The test suite PASSES now.

**Comment by RosebuDAO** [comment]
@ChaituVR it passes without errors now. sorry for wasting your time with those earlier revisions.

**Comment by will-holley** [comment]
This strategy is not officially affiliated with CityDAO and **should not imply as such**. Please remove "CityDAO" from its name so that people are not misled.

**Comment by RosebuDAO** [comment]
> This strategy is not officially affiliated with CityDAO and **should not imply as such**. Please remove "CityDAO" from its name so that people are not misled.

I'm a CityDAO Citizen, same as you. Once it's merged in I intend on creating a proposal to start using it, which would make it official. I'll rename it to citydao-one-human-one-vote as a compromise.

**Comment by RosebuDAO** [comment]
Yay I finally found an address that's both in the Proof of Humanity registry and owns a CityDAO Citizen NFT! I updated examples.json accordingly. It's this address: 0x99A5A6010e81d3Efa13d5832043E9C3B6c11Ef3d

Rather than wait for this to get merged I'll just propose it as soon as your [QV proposal](https://snapshot.org/#/daocity.eth/proposal/0x03fa9dd24efa6b64f3f05b1b55b794d9c8b2de98ee82f67d196050c7b54c6bca) ends, May 23rd :). That would make it "official".

**Comment by RosebuDAO** [comment]
You're right I may have jumped the gun, so I went ahead and started the [Proposal on CityDAO's forum](https://forum.citydao.io/t/cip-60-one-human-one-vote/1408)... since it might take a while to get "20 likes" before it goes up for official voting on Snapshot.

Let's just leave this PR right here in this state until the vote happens.

> The voting strategy that you're proposing has not been commissioned, audited, tested...

Bug reports and patches welcomed.

**Comment by ChaituVR** [comment]
Closing this as stale, feel free to reopen if needed

**Pull request closed without merging**
